language/move-vm/runtime/src/session.rs

```diff
@@ -82,116 +75,29 @@ impl<'r, 'l, S: MoveResolver> Session<'r, 'l, S> {
             &mut self.data_cache,
             gas_status,
             &mut self.native_extensions,
+            bypass_visibility,
         )
     }
 
-    /// Execute `module`::`fuction_name`<`ty_args`>(`args`) and return the effects in
-    /// an `ExecutionResult`, including
-    /// * the write set and events
-    /// * return values of the function
-    /// * changes to values passes by mutable reference to the function
-    /// Arguments to the function in `args` can be any type--ground types, user-defined struct
-    /// types, and references (including mutable references).
-    /// A reference argument in `args[i]` with type `&T` or `&mut T` will be deserialized as a `T`.
-    /// Pure arguments are deserialized in the obvious way.
-    ///
-    /// NOTE: The ability to deserialize `args` into arbitrary types is very powerful--e.g., it can
-    /// used to manufacture `signer`'s or `Coin`'s from raw bytes. It is the respmsibility of the
-    /// caller (e.g. adapter) to ensure that this power is useed responsibility/securely for its use-case.
-    pub fn execute_function_for_effects<V>(
-        self,
-        module: &ModuleId,
-        function_name: &IdentStr,
-        ty_args: Vec<TypeTag>,
-        args: Vec<V>,
-        gas_status: &mut GasStatus,
-    ) -> ExecutionResult
-    where
-        V: Borrow<[u8]>,
```

Comment:
This was done to prevent copies in performance-sensitive adapters (see https://github.com/diem/move/pull/136 for more details). We should be careful to preserve that.

Reply:
ah, didn't see this wasn't done on execute function and other spots (its all over in the runtime)
Will fix